synapse/replication/tcp/external_cache.py

```diff
@@ -72,23 +86,25 @@ async def set(self, cache_name: str, key: str, value: Any, expiry_ms: int) -> No
 
         logger.debug("Caching %s %s: %r", cache_name, key, encoded_value)
 
-        return await make_deferred_yieldable(
-            self._redis_connection.set(
-                self._get_redis_key(cache_name, key),
-                encoded_value,
-                pexpire=expiry_ms,
+        with response_timer.labels("set").time():
```

Comment:
Would it make sense to add the `cache_name` as a label?

Reply:
I don't think so, the times shouldn't depend on what we're caching (and I'm a bit cautious about the fact it adds a bunch of extra metrics). Broadly these should be super quick, and if they're not then something is wrong/congested.